dandi/models.py

```diff
@@ -791,6 +849,12 @@ class BareAssetMeta(CommonModel):
     wasAttributedTo: List[Participant] = Field(
         None, description="Participant(s) to which this file belongs to", nskey="prov"
     )
+    wasGeneratedBy: Optional[List[Union[Session, Project, Activity]]] = Field(
```

Comment:
This overrides the `CommonModel.wasGeneratedBy` attribute that is currently set in `nwb2asset()`.  Perhaps you should pick a different name?

Reply:
it's the same field and should be converged as necessary.

Reply:
The field in the parent class is a scalar, while this is a list.  That's bad inheritance.

Reply:
then the parent class should be adjusted to be a list. a given dandiset or asset could have been generated as as function of many activities. especially since we don't parcellate out wasGeneratedBy to specific elements.

Reply:
i adjusted the parent class.